### PR TITLE
✨ Deezer Rewrite

### PR DESCRIPTION
**Pull request by StrikerFRFX**
Screenshots:
mD77fxg8wO
wTT24P8DUL
x8XF200L5H

### CODE REVIEW

**Comment by StrikerFRFX** [comment]
Also Podcasts:
ioWvmLfkbT
CDyGt5BIvU

File: websites/D/Deezer/presence.ts [hunk 1]
```diff
@@ -7,129 +7,164 @@ const presence = new Presence({
...
-      parseInt(splitAudioTime[0]) * 60 + parseInt(splitAudioTime[1]),
-    parsedAudioDuration =
-      parseInt(splitAudioDuration[0]) * 60 + parseInt(splitAudioDuration[1]),
-    startTime = Date.now(),
-    endTime =
-      Math.floor(startTime / 1000) - parsedAudioTime + parsedAudioDuration;
-  return [Math.floor(startTime / 1000), endTime];
+function parseTimestamp(time: string) {
+  return parseInt(time.split(":")[0]) * 60 + parseInt(time.split(":")[1]);
 }
```
Comment (Bas950) [inline websites/D/Deezer/presence.ts:12]: pretty sure there is a presence. function for this

File: websites/D/Deezer/presence.ts [hunk 2]
```diff
@@ -7,129 +7,164 @@ const presence = new Presence({
...
-    presence.clearActivity();
+    const pathname = document.location.pathname;
+
+    if (pathname.includes("shows")) {
+      const data: PresenceData = {
+        details: "Browsing...",
+        state: "Shows",
+        largeImageKey: "deezer"
+      };
+      presence.setActivity(data);
```
Comment (Bas950) [inline websites/D/Deezer/presence.ts:98]: just use one and setactivity it at the end

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/D/Deezer/presence.ts (hunk 1 above)
Comment (StrikerFRFX) [reply websites/D/Deezer/presence.ts:12]: nope, it's a function made to specifically parse the audio timestamps since they're in a mm:ss format

Comment (Bas950) [reply websites/D/Deezer/presence.ts:12]: image

Comment (Bas950) [reply websites/D/Deezer/presence.ts:12]: presence.timestampFromFormat should be perfect for this.

Comment (StrikerFRFX) [reply websites/D/Deezer/presence.ts:12]: Alright will change it, thanks!

File: websites/D/Deezer/presence.ts [hunk 3]
```diff
@@ -92,35 +85,25 @@ presence.on("UpdateData", async () => {
...
-      data.state = "User's Loved";
+      (data.details = "Browsing..."), (data.state = "User's Loved");
     } else if (pathname.includes("playlists")) {
-      data.details = "Browsing...",
-      data.state = "User's Playlists";
+      (data.details = "Browsing..."), (data.state = "User's Playlists");
     } else if (pathname.includes("albums")) {
-      data.details = "Browsing...",
-      data.state = "User's Albums";
+      (data.details = "Browsing..."), (data.state = "User's Albums");
```
Comment (Bas950) [inline websites/D/Deezer/presence.ts]: what is  was before is better.

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by ririxidev — APPROVED** [review]

**Pull request merged**